third_party/amd/lib/TritonAMDGPUToLLVM/TargetInfo.cpp

```diff
@@ -52,6 +51,8 @@ Value printfPromoteValue(ConversionPatternRewriter &rewriter, Value value) {
 }
 } // namespace
 
+int TargetInfo::getSharedMemorySize() const { return 64 * 1024; }
```

Comment:
I'm wondering if we can query this size from frontend (like [this code](https://github.com/openai/triton/blob/ce7346f0471cd32f69225ccef7c8f72c0d2bd2b3/python/triton/compiler/compiler.py#L370)) and pass the information here? Therefore, we don't need to change it for the next generation AMD GPU.

Reply:
Generally there are two buckets of such information--one regarding microarchitecture and within a single compute unit, one regarding the final chip / SKU per the market slicing. for the former we typically put in the compilers as it's not a lot of variants and not frequently updating (on arch update every one or two years) so not a lot of overhead to update. they are also very important to compilation flow (e.g., max register count, max shared memory limit) and we sort of wanting static data baked into the compiler. for the latter category there are lots of variants and can happen anytime so better to query from the runtime, e.g., number of compute units on the hardware. (we can also put in the compiler for commonly seen ones for sure.)

Reply:
> so not a lot of overhead to update

I'm thinking about the situation where we need to support two archs with different SMEM sizes. 

> we sort of wanting static data baked into the compiler

If we query the information before the lowering of the IRs, can we still bake it as a static number?

Reply:
> I'm thinking about the situation where we need to support two archs with different SMEM sizes.

Yup the `TargetInfo` has the `gfx*` target string; we can switch on it to return different values. It's just right now we have the same value for various `gfx*` targets.

> If we query the information before the lowering of the IRs, can we still bake it as a static number?

True. Note that my previous comment is a bit general--wanting to call out why we bake in some information about the microarch in the compiler; not necessarily meaning shared memory limit is not static if plumbing through the runtime.

There is an issue here though if done that way--it means you must have the runtime right? meaning your dev environment must be the GPU you are compiling for. So it rules out cases where I'm developing a kernel for a CDNA GPU on a RNDA dev machine. (again this is generally about the split; not specifically for shared memory limit where we have no difference right now.) You cannot cross compile at all. I don't know whether this is something we'll support in triton; but to me seems better to avoid restricting it early.

Reply:
make sense. cross compilation is indeed a concern here.